pkg/noun/jets/c/aor.c

```diff
@@ -6,87 +6,70 @@
+  u3_noun
+  u3qc_aor(u3_noun a,
+           u3_noun b)
   {
-    if ( c3y == u3r_sing(a, b) ) {
-      return c3y;
-    }
-    else {
-      if ( c3y == u3ud(a) ) {
-        if ( c3y == u3ud(b) ) {
-          c3_w len_a_w, len_b_w;
-          c3_w *a_words, *b_words;
-          c3_y a_y, b_y;
-          if ( c3y == u3a_is_cat(a) ) {
-            len_a_w = 1;
-            a_words = &a;
-          }
-          else {
-            u3a_atom* a_u = u3a_to_ptr(a);
-            len_a_w = (a_u->len_w);
-            a_words = a_u->buf_w;
-          }
-          if ( c3y == u3a_is_cat(b) ) {
-            len_b_w = 1;
-            b_words = &b;
+    while ( 1 ) {
+      if ( c3y == u3r_sing(a, b) ) {
+        return c3y;
+      }
+      else {
+        if ( c3y == u3ud(a) ) {
+          if ( c3y == u3ud(b) ) {
+            c3_w len_a_w, len_b_w;
+            c3_w *a_words, *b_words;
+            c3_y a_y, b_y;
+            if ( c3y == u3a_is_cat(a) ) {
+              len_a_w = 1;
+              a_words = &a;
+            }
+            else {
+              u3a_atom* a_u = u3a_to_ptr(a);
+              len_a_w = (a_u->len_w);
+              a_words = a_u->buf_w;
+            }
+            if ( c3y == u3a_is_cat(b) ) {
+              len_b_w = 1;
+              b_words = &b;
+            }
+            else {
+              u3a_atom* b_u = u3a_to_ptr(b);
+              len_b_w = (b_u->len_w);
+              b_words = b_u->buf_w;
+            }
+            c3_w len_min = c3_min(len_a_w, len_b_w);
+            for (c3_w i_w = 0; i_w < len_min; i_w++) {
+              for (c3_w j = 0; j < 4; j++) {
+                a_y = (a_words[i_w] >> (8*j)) & 0xFF;
+                b_y = (b_words[i_w] >> (8*j)) & 0xFF;
+                if ( a_y != b_y ) return __(a_y < b_y);
+              }
+            }
+            return __(len_a_w < len_b_w);
           }
           else {
-            u3a_atom* b_u = u3a_to_ptr(b);
-            len_b_w = (b_u->len_w);
-            b_words = b_u->buf_w;
-          }
-          c3_w len_min = c3_min(len_a_w, len_b_w);
-          for (c3_w i_w = 0; i_w < len_min; i_w++) {
-            for (c3_w j = 0; j < 4; j++) {
-              a_y = (a_words[i_w] >> (8*j)) & 0xFF;
-              b_y = (b_words[i_w] >> (8*j)) & 0xFF;
-              if ( a_y != b_y ) return __(a_y < b_y);
-            }
+            return c3y;
           }
-          return __(len_a_w < len_b_w);
-        }
-        else {
-          return c3y;
-        }
-      }
-      else {
-        if ( c3y == u3ud(b) ) {
-          return c3n;
         }
         else {
-          if ( c3y == u3r_sing(u3h(a), u3h(b)) ) {
-            return 3;
+          if ( c3y == u3ud(b) ) {
+            return c3n;
           }
           else {
-            return 2;
+            if ( c3y == u3r_sing(u3h(a), u3h(b)) ) {
+              a = u3t(a);
+              b = u3t(b);
+            }
+            else {
+              a = u3h(a);
+              b = u3h(b);
+            }
```

Comment:
i think this would be easier to follow if the non-atom cases came first, like the hoon.

Reply:
done, the code flow now follows ++aor definition in hoon